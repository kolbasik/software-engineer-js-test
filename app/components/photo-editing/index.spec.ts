import {
	PhotoEditingState,
	photoEditingReducer,
	movePhoto,
	scalePhoto,
} from './index';

describe('PhotoEditing', () => {
	// Arrange
	const previousState: PhotoEditingState = {
		project: {
			id: 'TEST',
			canvas: {width: 5, height: 10},
			image: {src: 'none', width: 50, height: 100},
			matrix: {a: 1, b: 0, c: 0, d: 1, e: 0, f: 0},
		},
	};

	describe('movePhoto', () => {
		it('should be handled', () => {
			// Act
			const nextState = photoEditingReducer(
				previousState,
				movePhoto({dx: -5, dy: -3}),
			);

			// Assert
			expect(nextState).not.toBe(previousState);
			expect(nextState).toMatchSnapshot();
		});
	});

	describe('scalePhoto', () => {
		it('should be handled', () => {
			// Act
			const nextState = photoEditingReducer(
				previousState,
				scalePhoto({scale: 2}),
			);

			// Assert
			expect(nextState).not.toBe(previousState);
			expect(nextState).toMatchSnapshot();
		});
	});
});
