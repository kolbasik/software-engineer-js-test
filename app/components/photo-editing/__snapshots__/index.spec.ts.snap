// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoEditing movePhoto should be handled 1`] = `
Object {
  "project": Object {
    "canvas": Object {
      "height": 10,
      "width": 5,
    },
    "id": "TEST",
    "image": Object {
      "height": 100,
      "src": "none",
      "width": 50,
    },
    "matrix": Object {
      "a": 1,
      "b": 0,
      "c": 0,
      "d": 1,
      "e": -5,
      "f": -3,
    },
  },
}
`;

exports[`PhotoEditing scalePhoto should be handled 1`] = `
Object {
  "project": Object {
    "canvas": Object {
      "height": 10,
      "width": 5,
    },
    "id": "TEST",
    "image": Object {
      "height": 100,
      "src": "none",
      "width": 50,
    },
    "matrix": Object {
      "a": 2,
      "b": 0,
      "c": 0,
      "d": 2,
      "e": -2.5,
      "f": -5,
    },
  },
}
`;
